lib/web/join_tokens.go

```diff
@@ -252,12 +262,22 @@ func (h *Handler) getAppJoinScriptHandle(w http.ResponseWriter, r *http.Request,
 		return nil, nil
 	}
 
+	var autoUpgradesVersion string
+	autoUpgrades := automaticUpgrades(h.ClusterFeatures)
+	if autoUpgrades {
+		autoUpgradesVersion, err = h.cfg.AutomaticUpgradesChannels.DefaultVersion(r.Context())
+		if err != nil {
+			return nil, trace.Wrap(err, "failed to get auto-upgrade version")
+		}
+	}
```

Comment:
Can we extract this to a method? We are re-using the same logic in multiple places.